driver/src/orderbook/mod.rs

```diff
@@ -46,9 +50,26 @@ impl<'a> PaginatedStableXOrderBookReader<'a> {
 
 impl<'a> StableXOrderBookReading for PaginatedStableXOrderBookReader<'a> {
     fn get_auction_index(&self) -> Result<U256> {
-        self.contract
-            .get_current_auction_index()
-            .map(|batch_collecting_orders| batch_collecting_orders - 1)
+        let batch_collecting_orders = self.contract.get_current_auction_index()?;
+        let batch_to_solve = batch_collecting_orders - 1;
+
+        Ok(U256::from(batch_to_solve))
+    }
+
+    fn get_auction_remaining_solution_time(&self) -> Result<Duration> {
+        // NOTE: Solutions are due 1 minutes before the batch ends and we give
+        //   ourselves another 30 seconds for submitting the solution.
+        const SOLUTION_DEADLINE: Duration = Duration::from_secs(90);
+
+        let time_remaining = {
+            let seconds = self.contract.get_current_auction_remaining_seconds()?;
+            Duration::from_secs(seconds.as_u64())
+        };
+        let solution_time_remaining = time_remaining
+            .checked_sub(SOLUTION_DEADLINE)
```

Comment:
nit: saturating sub https://docs.rs/ethereum-types/0.3.1/ethereum_types/struct.U256.html#method.saturating_sub

Reply:
`time_remaining` is a `Duration` which does not have a `saturating_sub` :cry: